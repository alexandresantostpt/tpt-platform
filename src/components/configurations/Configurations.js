import PropTypes from 'prop-types'
import React from 'react'

import { connect } from 'react-redux'

import { i18n } from '@i18n'

import Dropdown from '@components/dropdown/Dropdown'
import DropdownItem from '@components/dropdown/item/DropdownItem'
import DropdownLabel from '@components/dropdown/label/DropdownLabel'
import DropdownTitle from '@components/dropdown/title/DropdownTitle'

import { requestLogoutAction } from '@features/login/actions'

import { navigateTo } from '@utils/browser'
import { routesRedefinePassword } from '@features/login/routes'

const Configurations = ({ requestLogoutAction: handleLogoutAction }) => {
    const { path } = routesRedefinePassword

    return (
        <Dropdown icon="config">
            <DropdownItem>
                <DropdownTitle>{i18n.t('titles.configurations')}</DropdownTitle>
            </DropdownItem>
            <DropdownItem bigger onClick={() => navigateTo(path)}>
                <DropdownLabel dark>{i18n.t('titles.changePassword')}</DropdownLabel>
            </DropdownItem>
            <DropdownItem bigger center onClick={() => handleLogoutAction()}>
                <DropdownLabel dark>{i18n.t('labels.exit')}</DropdownLabel>
            </DropdownItem>
        </Dropdown>
    )
}

Configurations.propTypes = {
    requestLogoutAction: PropTypes.func.isRequired
}

const mapDispatchToProps = {
    requestLogoutAction
}

export default connect(
    null,
    mapDispatchToProps
)(Configurations)
