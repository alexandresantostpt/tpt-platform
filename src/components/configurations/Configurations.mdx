---
name: Configurations
route: /configurations
---
import { Playground, PropsTable } from 'docz'

import Configurations from './Configurations'

import '@cfg/scripts'
import '@cfg/style'

# Configurations
<PropsTable of={ Configurations }/>

## Basic usage

```
<Configurations/>
```

<Playground>
    <Configurations/>
</Playground>
