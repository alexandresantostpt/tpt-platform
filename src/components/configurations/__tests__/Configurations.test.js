import configureMockStore from 'redux-mock-store'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import Configurations from '../Configurations'

describe('Tests for Configurations component', () => {
    let mockStore = {}
    let store = {}

    beforeEach(() => {
        mockStore = configureMockStore()
        store = mockStore()
    })

    it('Should render without crash', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Configurations />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
