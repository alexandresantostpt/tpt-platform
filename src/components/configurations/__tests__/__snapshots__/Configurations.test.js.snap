// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Configurations component Should render without crash 1`] = `
<div
  className="Dropdown__StyledDropdown-sc-14kwj8x-0 iZGQRX"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="icon icon-config Icon__StyledIcon-crr1cg-0 cmjVJQ"
    disabled={false}
    fontSize="1.75rem"
    onClick={null}
  />
  <ul
    className="DropdownList__StyledDropdownList-vjd4uo-0 iDpmEs"
  >
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 jhIdjc"
      onClick={null}
    >
      <span
        className="DropdownTitle__StyledDropdownTitle-y9o5oh-0 hFdNKe"
      >
        Configurações
      </span>
    </li>
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 kBLyiT"
      onClick={[Function]}
    >
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
      >
        Alterar senha
      </span>
    </li>
    <li
      className="DropdownItem__StyledDropdownItem-sc-102yxuy-0 hOZFkb"
      onClick={[Function]}
    >
      <span
        className="DropdownLabel__StyledDropdownLabel-scn6ga-0 fJmcGN"
      >
        Sair
      </span>
    </li>
  </ul>
</div>
`;
